src/pc/render_shared.c

```diff
@@ -0,0 +1,98 @@
+#include <common.h>
+#include <log.h>
+#include <game.h>
+#include <stdlib.h>
+#include "pc.h"
+#include <assert.h>
+#include "sdl_defs.h"
+
+enum Renderers render_mode = RENDER_SOFT;
+
+int SoftDrawSync(int mode);
+DISPENV* SoftPutDispEnv(DISPENV* env);
+void SoftSetDrawEnv(DR_ENV* dr_env, DRAWENV* env);
+DRAWENV* SoftPutDrawEnv(DRAWENV* env);
+void SoftDrawOTag(OT_TYPE* p);
+
+int GlDrawSync(int mode);
+DISPENV* GlPutDispEnv(DISPENV* env);
+void GlSetDrawEnv(DR_ENV* dr_env, DRAWENV* env);
+DRAWENV* GlPutDrawEnv(DRAWENV* env);
+void GlDrawOTag(OT_TYPE* p);
+
+int MyDrawSync(int mode) {
+    if (render_mode == RENDER_SOFT) {
```

Comment:
Not for this PR, but I think we could just do something like the following later on:

```c
typedef struct {
    int (*DrawSync)(int mode);
    ...
} Renderer;

static Renderer renderer_hw = {
    .DrawSync = GlDrawSync,
    ...
};
static Renderer renderer_sw = {
    .DrawSync = SoftDrawSync,
    ...
};

static Renderer* renderer = &renderer_sw;
int MyDrawSync(int mode) {
    return renderer->DrawSync(mode);
}

```